modules/table/src/main/java/org/apache/ignite/internal/table/distributed/PartitionMover.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed;
+
+import static org.apache.ignite.internal.utils.RebalanceUtil.recoverable;
+import static org.apache.ignite.lang.ErrorGroups.Common.NODE_STOPPING_ERR;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.util.IgniteSpinBusyLock;
+import org.apache.ignite.lang.IgniteInternalException;
+import org.apache.ignite.lang.NodeStoppingException;
+import org.apache.ignite.raft.client.Peer;
+import org.apache.ignite.raft.client.service.RaftGroupService;
+
+/**
+ * Class for moving partitions.
+ */
+public class PartitionMover {
```

Comment:
This change is required for the following reason: this class replaces a parameter that was declared as a `BiFunction<Collection<Peer>, Long, CompletableFuture<Void>>`. Since I'm introducing a new parameter to the function call, this type would have been changed to something like `TriFunction<Collection<Peer>, Collection<Peer>, Long, CompletableFuture<Void>>`. Do you really think this is a viable alternative? I can leave everything as is if it will make review faster and easier for the reviewer.

Reply:
Also, because of the previous bad encapsulation, we had a test that ran for 10 seconds instead of having a single unit test that runs for a couple of milliseconds.

Reply:
Oh, I see, make sense.